intelmq/bots/outputs/smtp_batch/output.py

```diff
@@ -0,0 +1,349 @@
+            except TimeoutError:
+                print(f"Trying again: {mail_record}... ", flush=True)
+                for s in range(1, 4):
+                    time.sleep(s)
+                    try:
+                        messages = self.cache.redis.lrange(mail_record, 0, -1)
+                        print("... Success!", flush=True)
+                        break
+                    except TimeoutError:
+                        print("... failed ...", flush=True)
+                        continue
+                else:
+                    # visible both warning and print
+                    print(f"Warning: {mail_record} timeout, too big to read from redis", flush=True)
+                    self.logger.warning(f"Warning: {mail_record} timeout, too big to read from redis")
+                    self.timeout.append(mail_record)
+                    continue
+
+            lines.extend(json.loads(str(message, encoding="utf-8")) for message in messages)
+
+            # prepare rows for csv attachment
+            threshold = datetime.datetime.now() - datetime.timedelta(
+                days=self.ignore_older_than_days) if getattr(self, 'ignore_older_than_days',
+                                                             False) else False
+            fieldnames = set()
+            rows_output = []
+            for row in lines:
+                if threshold and row["time.observation"][:19] < threshold.isoformat()[:19]:
+                    continue
+                fieldnames = fieldnames | set(row.keys())
+                keys = set(allowed_fieldnames).intersection(row)
+                ordered_keys = []
+                for field in allowed_fieldnames:
+                    if field in keys:
+                        ordered_keys.append(field)
+                try:
+                    row["raw"] = b64decode(row["raw"]).decode("utf-8").strip().replace("\n", r"\n").replace("\r", r"\r")
+                except (ValueError, KeyError):  # not all events have to contain the "raw" field
+                    pass
+                rows_output.append(OrderedDict({fieldnames_translation[k]: row[k] for k in ordered_keys}))
+
+            # prepare headers for csv attachment
+            ordered_fieldnames = []
+            for field in allowed_fieldnames:
+                ordered_fieldnames.append(fieldnames_translation[field])
+
+            # write data to csv
+            output = StringIO()
+            dict_writer = csv.DictWriter(output, fieldnames=ordered_fieldnames)
+            dict_writer.writerow(dict(zip(ordered_fieldnames, ordered_fieldnames)))
+            dict_writer.writerows(rows_output)
+
+            email_to = str(mail_record[len(self.key):], encoding="utf-8")
+            count = len(rows_output)
+            if not count:
+                path = None
+            else:
+                filename = f'{time.strftime("%y%m%d")}_{count}_events'
+                path = self.TMP_DIR + filename + '_' + email_to + '.zip'
```

Comment:
Totally agreed! Since I see no security risk, I used bare base64 (ok?).

```python
path = self.TMP_DIR + filename + '_' + b64encode(email_to.encode("utf-8")).decode("utf-8") + '.zip'
```

Reply:
Please use [`tempfile`](https://docs.python.org/3/library/tempfile.html) for temporary files and directories.